esiclient/v1/node_network.py

```diff
@@ -214,27 +214,39 @@ def take_action(self, parsed_args):
                 node.name, port.name, mac_string))
             ironic_client.node.vif_attach(node_uuid, port.id, **vif_info)
         elif parsed_args.network:
-            print("Attaching network {1} to node {0}{2}".format(
-                node.name, network.name, mac_string))
-            port_name = utils.get_port_name(network.name, prefix=node.name)
-            port = utils.get_or_create_port(port_name, network, neutron_client)
-            ironic_client.node.vif_attach(node_uuid, port.id, **vif_info)
-            port = neutron_client.get_port(port.id)
+            if network is not None:
+                print("Attaching network {1} to node {0}{2}".format(
+                    node.name, network.name, mac_string))
+                port_name = utils.get_port_name(network.name, prefix=node.name)
+                port = utils.get_or_create_port(port_name, network,
+                                                neutron_client)
+                ironic_client.node.vif_attach(node_uuid, port.id, **vif_info)
+                port = neutron_client.get_port(port.id)
+            else:
+                print("The specified network does not exist."
+                      "Please provide a valid network name.")
```

Comment:
Instead of printing an error message, you'll want to raise `exceptions.CommandError` (you can find examples of this if you grep in the repository). You should also check if the network exists right away - up in line 177.